packages/flame_bloc/lib/src/flame_bloc_listener.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:bloc/bloc.dart';
+import 'package:flame/components.dart';
+
+import '../flame_bloc.dart';
+
+/// {@template flame_bloc_listener}
+/// A [Component] which can listen to changes in a [Bloc] state.
+/// {@endtemplate}
+class FlameBlocListener<B extends BlocBase<S>, S> extends Component
+    with FlameBlocListenable<B, S> {
+  /// {@macro flame_bloc_listener}
+  FlameBlocListener({
+    B? bloc,
+    required void Function(S state) onNewState,
+    bool Function(S previousState, S newState)? listenWhen,
```

Comment:
It would be better to have these types as `typedefs` since we are repeating them here and in the field definitions.